streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPrint.java

```diff
@@ -62,13 +62,19 @@ public void init(ProcessorContext context) {
             if (valueSerde == null) {
                 this.valueSerde = context.valueSerde();
             }
-        }
 
-        @Override
-        public void process(final K key, final V value) {
-            final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
-            final V deValue = (V) maybeDeserialize(value, valueSerde.deserializer());
-            action.apply(deKey, deValue);
+            // no mapper provided, using a default one which takes care of checking byte[] for deserialization
+            if (mapper == null) {
+                mapper = new KeyValueMapper<K, V, String>() {
+                    @Override
+                    public String apply(K key, V value) {
+                        final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
```

Comment:
I think they only way we can avoid the `instanceof` is if we just use `toString` on every object. In which case a `byte[]` is just going to print out the array address rather than anything useful.